plugin/plugin.go

```diff
@@ -57,8 +57,7 @@ func run(dockerClient *docker.Client, weave *weaveapi.Client, address, meshAddre
 	if !isPluginV2 {
 		Log.Println("Creating default 'weave' network")
 		options := map[string]interface{}{MulticastOption: "true"}
-		// TODO: the driver name should be extracted from pluginMeshSocket
-		dockerClient.EnsureNetwork("weave", "weavemesh", defaultSubnet, options)
+		dockerClient.EnsureNetwork("weave", pluginNameFromAddress(meshAddress), defaultSubnet, options)
```

Comment:
Not related to your PR, but this should be moved to the `if meshAddress != '' {` branch.